Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +229,41 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (index < 0 || index > array?.Length)
+            {
+                return array;
+            }
+            if (ArrayIsNullOrEmpty(array))
+            {
+                return new int[] { number };
+            }
+            
+            if (index == 0)
+            {
+                return InsertFirst(array, number);
+            }
+            if (index == array.Length)
+            {
+                return InsertLast(array, number);
+            }
+
+            int[] auxArray = new int[array.Length + 1];
+
+            for (int i = 0, j = 0, k = 0; i < auxArray.Length; i++)
+            {
+                if (j == index)
+                {
+                    auxArray[j] = number;
+                    j++;
+                    continue;
+                }
+                auxArray[j] = array[k];
+                j++;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not update the loop counter 'j' within the loop body.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4930349)